modules/car/view/record.inc

```diff
@@ -0,0 +1,62 @@
+<?
+$primarykey['CompanyCarID'] = $CompanyCarID;
+
+if($_lib['input']->getProperty('action_car_update')) {
+  #Possible to extend or alter parameters here
+  #print_r($_POST);
+  #print_r($primarykey);
+  for($milage_year = $car_purchase_year; $milage_year <= $_year; $milage_year++) {
+    $post_values = array("companycarmilage_km0101" => $_POST["companycarmilage_km0101_$milage_year"], "companycarmilage_km3112" => $_POST["companycarmilage_km3112_$milage_year"]);
+    $_lib['db']->db_update_hash($post_values, 'companycarmilage', array("CompanyCarID" => $CompanyCarID, "MilageYear" => $milage_year));
+  }
+  $_lib['db']->db_update_hash($_POST, $db_table, $primarykey);
+}
+elseif($_lib['input']->getProperty('action_car_new'))
+{
+
+	if($_lib['input']->getProperty('companycar_CompanyCarID') > 0) {
+
+		$query  = "select CompanyCarID from $db_table where CompanyCarID=" . (int) $_lib['input']->getProperty('companycar_CompanyCarID') . "";
+		$result = $_lib['db']->db_query($query);
+		if($_lib['db']->db_numrows($result) == 0)
+		{
+			$post = $_POST;
+			$post['companycar_ValidFrom'] = "NOW()";
+			$post['companycar_Active']  = 1;
+
+			$query  = "select CompanyCarID from $db_table where CompanyCarID=" . (int) $_lib['input']->getProperty('companycar_CompanyCarID') . "";
+			$_lib['db']->db_insert2(array('query'=>"insert into companycar (CompanyCarID, ValidFrom, Active) values (". (int) $post['companycar_CompanyCarID'].", ". $post['companycar_ValidFrom'] .", '". (int) $post['companycar_Active']."')"));
+			$CompanyCarID = $_POST['companycar_CompanyCarID'];
+		}
+		else
+		{
+			$CompanyCarID = $_POST['companycar_CompanyCarID'];
+		}
```

Comment:
Put `$CompanyCarID = $_POST['companycar_CompanyCarID'];` outside if block and remove else block.